src/theory/sets/theory_sets_private.cpp

```diff
@@ -672,123 +387,132 @@ void TheorySetsPrivate::fullEffortCheck(){
-              Trace("sets-mem") << it2->first << " ";
+            for (const std::pair<const Node, Node>& it2 : smem)
+            {
+              Trace("sets-mem") << it2.first << " ";
             }
           }
-          std::map< Node, Node >::iterator its = d_eqc_singleton.find( s );
-          if( its!=d_eqc_singleton.end() ){
-            Trace("sets-mem") << " : Singleton : " << its->second;
+          Node ss = d_state.getSingletonEqClass(s);
+          if (!ss.isNull())
+          {
+            Trace("sets-mem") << " : Singleton : " << ss;
           }
           Trace("sets-mem") << std::endl;
         }
       }
-      checkSubtypes( lemmas );
-      flushLemmas( lemmas, true );
-      if( !hasProcessed() ){
-      
-        checkDownwardsClosure( lemmas );
+      checkSubtypes();
+      d_im.flushPendingLemmas(true);
+      if (!d_im.hasProcessed())
+      {
+        checkDownwardsClosure();
         if( options::setsInferAsLemmas() ){
-          flushLemmas( lemmas );
+          d_im.flushPendingLemmas();
         }
-        if( !hasProcessed() ){
-          checkUpwardsClosure( lemmas );
-          flushLemmas( lemmas );
-          if( !hasProcessed() ){
-            checkDisequalities(lemmas);
-            flushLemmas(lemmas);
-            if (!hasProcessed() && d_card_enabled)
+        if (!d_im.hasProcessed())
+        {
+          checkUpwardsClosure();
+          d_im.flushPendingLemmas();
+          if (!d_im.hasProcessed())
+          {
+            checkDisequalities();
+            d_im.flushPendingLemmas();
+            if (!d_im.hasProcessed() && d_card_enabled)
             {
-              // for cardinality
-              checkCardBuildGraph( lemmas );
-              flushLemmas( lemmas );
-              if( !hasProcessed() ){
-                checkMinCard( lemmas );
-                flushLemmas( lemmas );
-                if( !hasProcessed() ){
-                  checkCardCycles( lemmas );
-                  flushLemmas( lemmas );
-                  if( !hasProcessed() ){
-                    std::vector<Node> intro_sets;
-                    checkNormalForms( lemmas, intro_sets );
-                    flushLemmas( lemmas );
-                    if (!hasProcessed() && !intro_sets.empty())
-                    {
-                      Assert(intro_sets.size() == 1);
-                      Trace("sets-intro")
-                          << "Introduce term : " << intro_sets[0] << std::endl;
-                      Trace("sets-intro") << "  Actual Intro : ";
-                      debugPrintSet(intro_sets[0], "sets-nf");
-                      Trace("sets-nf") << std::endl;
-                      Node k = getProxy(intro_sets[0]);
-                      d_sentLemma = true;
-                    }
-                  }
-                }
-              }
+              // call the check method of the cardinality solver
+              d_cardSolver->check();
```

Comment:
What is the rationale of removing all occurrences of `flushLemmas( lemmas );` in the code block above from the function `d_cardSolver::check`?